src/Compilers/CSharp/Test/Emit/CodeGen/IndexAndRangeTests.cs

```diff
@@ -16,6 +16,267 @@ private CompilationVerifier CompileAndVerifyWithIndexAndRange(string s, string e
             return CompileAndVerify(comp, expectedOutput: expectedOutput);
         }
 
+        [Fact]
+        public void PatternIndexList()
+        {
+            var src = @"
+using System;
+using System.Collections.Generic;
+class C
+{
+    static void Main()
+    {
+        var list = new List<int>() { 2, 4, 5, 6 };
```

Comment:
Nit: Move the creation of the list to a different function and the IL diff below will be more targeted.